runtime/compiler/net/LoadSSLLibs.cpp

```diff
@@ -0,0 +1,537 @@
+   printf(" SSL_CTX_use_certificate %p\n", OSSL_CTX_use_certificate);
+   printf(" SSL_CTX_check_private_key %p\n", OSSL_CTX_check_private_key);
+   printf(" SSL_CTX_set_verify %p\n", OSSL_CTX_set_verify);
+   printf(" SSL_CTX_free %p\n", OSSL_CTX_free);
+   printf(" SSL_CTX_get_cert_store %p\n", OSSL_CTX_get_cert_store);
+
+   printf(" BIO_new_mem_buf %p\n", OBIO_new_mem_buf);
+   printf(" BIO_free_all %p\n", OBIO_free_all);
+   printf(" BIO_new_ssl %p\n", OBIO_new_ssl);
+   printf(" BIO_write %p\n", OBIO_write);
+   printf(" BIO_read %p\n", OBIO_read);
+
+   printf(" PEM_read_bio_PrivateKey %p\n", OPEM_read_bio_PrivateKey);
+   printf(" PEM_read_bio_X509 %p\n", OPEM_read_bio_X509);
+   printf(" PEM_X509_INFO_read_bio %p\n", OPEM_X509_INFO_read_bio);
+
+   printf(" X509_STORE_add_cert %p\n", OX509_STORE_add_cert);
+   printf(" X509_STORE_add_crl %p\n", OX509_STORE_add_crl);
+   printf(" X509_free %p\n", OX509_free);
+
+   printf(" ERR_print_errors_fp %p\n", OERR_print_errors_fp);
+
+   printf("=============================================================\n\n");
+   }
+
+bool loadLibsslAndFindSymbols()
+{
+   void *handle = NULL;
+
+   handle = loadLibssl();
+   if (!handle)
+      {
+      printf("\n===OpenSSL TEST: failed to load any libssl\n");
+      return false;
+      }
+
+   int ossl_ver = findLibsslVersion(handle);
+   if (-1 == ossl_ver)
+      {
+      unloadLibssl(handle);
+      return false;
+      }
+
+   //OOpenSSL_add_ssl_algorithms = (OOpenSSL_add_ssl_algorithms_t *)findLibsslSymbol(handle, "OpenSSL_add_ssl_algorithms");
+
+   if (0 == ossl_ver)
+   {
+   OOPENSSL_init_ssl = &OSSL102_OOPENSSL_init_ssl;
+
+   OSSL_load_error_strings = (OSSL_load_error_strings_t *)findLibsslSymbol(handle, "SSL_load_error_strings");
+   OSSL_library_init = (OSSL_library_init_t *)findLibsslSymbol(handle, "SSL_library_init");
+
+   OSSLv23_server_method = (OSSLv23_server_method_t *)findLibsslSymbol(handle, "SSLv23_server_method");
+   OSSLv23_client_method = (OSSLv23_client_method_t *)findLibsslSymbol(handle, "SSLv23_client_method");
+   }
+   else
+   {
+   OOPENSSL_init_ssl = (OOPENSSL_init_ssl_t *)findLibsslSymbol(handle, "OPENSSL_init_ssl");
+
+   OSSL_load_error_strings = &OSSL110_load_error_strings;
+   OSSL_library_init = &OSSL110_library_init;
+
+   OSSLv23_server_method = (OSSLv23_server_method_t *)findLibsslSymbol(handle, "TLS_server_method");
+   OSSLv23_client_method = (OSSLv23_client_method_t *)findLibsslSymbol(handle, "TLS_client_method");
+   }
+
+   OSSL_CTX_ctrl = (OSSL_CTX_ctrl_t *)findLibsslSymbol(handle, "SSL_CTX_ctrl");
+   if (0 == ossl_ver)
+      {
+      OSSL_CTX_set_ecdh_auto = &OSSL102_CTX_set_ecdh_auto;
+      }
+   else
+      {
+      OSSL_CTX_set_ecdh_auto = &OSSL110_CTX_set_ecdh_auto;
+      }
+
+   OBIO_ctrl = (OBIO_ctrl_t *)findLibsslSymbol(handle, "BIO_ctrl");
+
+   OSSL_CIPHER_get_name = (OSSL_CIPHER_get_name_t *)findLibsslSymbol(handle, "SSL_CIPHER_get_name");
```

Comment:
Do we need a certain order of these calls? if not we could group those methods that are different in version 1.0 vs v.1 and do the test `if (0 == ossl_ver)` only once